src/clients/mattermost_proxy_plugin.ts

```diff
@@ -0,0 +1,128 @@
+import fetch from 'node-fetch';
+
+import {Oauth2App} from 'src/utils';
+
+import {Routes, AppsPluginName, PathAPI, PathKV} from '../utils/constants';
+import {getManifest} from '../manifest';
+
+export interface ProxyClient {
+    kvSet(key: string, value: any): Promise<void>;
+    kvGet(key: string): Promise<any>;
+    kvDelete(key: string): void;
+    storeOauth2App(id: string, secret: string): void;
+    storeOauth2User(token: string): void
+}
+
+export const newProxyClient = (botAccessToken: string, baseURL: string): ProxyClient => {
+    return new ProxyClientImpl(botAccessToken, baseURL);
+};
+
+class ProxyClientImpl implements ProxyClient {
+    token: string
+    url: string
+
+    constructor(token: string, baseURL: string) {
+        this.token = token;
+        this.url = baseURL;
+    }
+    async kvSet(key: string, value: any): Promise<void> {
+        const url = this.url + this.kvPath(key);
+        try {
+            this.doAPIPost(url, value);
+        } catch (e) {
+            console.log('e', e);
+        }
+    }
```

Comment:
If we let the `doAPIPost` call resolve its promise before exiting this function, we can let callers of `kvSet` wait for the execution to be done by using `await kvSet()`. This also lets us be able to bubble an error from `doAPIPost` if necessary.

```suggestion
    async kvSet(key: string, value: any): Promise<void> {
        const url = this.url + this.kvPath(key);
        return this.doAPIPost(url, value);
    }
```

If you want to sanitize the error or change it, you can do that in a catch block and rethrow an error:

```go
try {
    return this.doAPIPost(url, value);
} catch (e) {
    throw new Error('setKv failed: ' + e.message);
}
```

Reply:
Done.